src/Composer/CustomDirectoryInstaller/PearPlugin.php

```diff
@@ -5,13 +5,23 @@
 use Composer\Composer;
 use Composer\IO\IOInterface;
 use Composer\Plugin\PluginInterface;
-use Composer\CustomDirectoryInstaller\PearInstaller;
 
 class PearPlugin implements PluginInterface
 {
   public function activate (Composer $composer, IOInterface $io)
   {
+    if (!class_exists('Composer\Composer\Installer\PearInstaller')) {
+      return;
+    }
     $installer = new PearInstaller($io, $composer);
     $composer->getInstallationManager()->addInstaller($installer);
   }
+
+  public function deactivate(Composer $composer, IOInterface $io)
```

Comment:
is there a reason why is it empty ?

Reply:
the deactivate method is only used in composer 2, which does not support a PearInstaller. so the installation won't and we don't need to have an uninstall method.
in case of composer 1 the uninstall won't be called.